test/Vault1Test.js

```diff
@@ -1,13 +1,38 @@
-const { expect } = require("chai");
-const { ethers } = require("hardhat");
+const {expect} = require("chai");
+const {ethers, waffle} = require("hardhat");
+const provider = waffle.provider;
 
 describe("Vault 1", () => {
-	beforeEach(async () => {
-		Vault1 = await ethers.getContractFactory("Vault1");
-		[owner] = await ethers.getSigners();
+    beforeEach(async () => {
+        [owner] = await ethers.getSigners();
 
-		vault1 = await Vault1.deploy();
+        DappCampToken = await ethers.getContractFactory("DappCampToken", owner);
+        dcToken = await DappCampToken.deploy(1e9);
+        await dcToken.deployed();
 
-		await vault1.deployed();
-	});
+        Vault1 = await ethers.getContractFactory("Vault1");
+        vault1 = await Vault1.deploy();
+        await vault1.deployed();
+    });
+
+    it("Deposit & withdrawal should work", async function () {
```

Comment:
Better to split these tests, so that when a test is failing you know which part of contract to look for and fix